tests/test_cli_options.py

```diff
@@ -0,0 +1,507 @@
+
+            #List of Strings Options:
+            # Options format:
+            #   (intlist/stringlist, MA step, long_flag, short_flag, test_value, default_value)
+            # The following options can be None:
+            #   short_flag
+            #   default_value
+            OptionStruct("intlist", "profile", "--batch-sizes", "-b", "2, 4, 6", "1"),
+            OptionStruct("intlist", "profile", "--concurrency", "-c", "1, 2, 3", None),
+            OptionStruct("stringlist", "profile", "--triton-docker-mounts", None, "a:b:c, d:e:f", None, extra_commands=["--triton-launch-mode", "docker"]),
+            OptionStruct("stringlist", "profile", "--gpus", None, "a, b, c", "all"),
+            OptionStruct("stringlist", "analyze", "--inference-output-fields", None, "a, b, c",
+                "model_name,batch_size,concurrency,model_config_path,instance_group,satisfies_constraints,perf_throughput,perf_latency_p99"),
+            OptionStruct("stringlist", "analyze", "--gpu-output-fields", None, "a, b, c",
+                "model_name,gpu_uuid,batch_size,concurrency,model_config_path,instance_group,satisfies_constraints,gpu_used_memory,gpu_utilization,gpu_power_usage"),
+            OptionStruct("stringlist", "analyze", "--server-output-fields", None, "a, b, c",
+                "model_name,gpu_uuid,gpu_used_memory,gpu_utilization,gpu_power_usage"),
+
+            # No OP Options:
+            # Option format:
+            # (noop, any MA step, long_flag,)
+            # These commands arent tested directly but are here to ensure that
+            # the count is correct for all options in the config.
+            # Some of these are required to run the subcommand
+            # Others are yaml only options
+            OptionStruct("noop", "profile", "--model-repository"),
+            OptionStruct("noop", "profile", "--profile-models"),
+            OptionStruct("noop", "analyze", "--analysis-models"),
+            OptionStruct("noop", "report", "--report-model-configs"),
+            OptionStruct("noop", "report", "--output-formats", "-o", ["pdf", "csv", "png"], "pdf", "SHOULD_FAIL"),
+            OptionStruct("noop", "yaml_profile", "constraints"),
+            OptionStruct("noop", "yaml_profile", "objectives"),
+            OptionStruct("noop", "yaml_profile", "triton_server_flags"),
+            OptionStruct("noop", "yaml_profile", "perf_analyzer_flags"),
+            OptionStruct("noop", "yaml_profile", "triton_docker_labels"),
+            OptionStruct("noop", "yaml_profile", "triton_server_environment"),
+            OptionStruct("noop", "yaml_analyze", "constraints"),
+            OptionStruct("noop", "yaml_analyze", "objectives"),
+            OptionStruct("noop", "yaml_analyze", "plots"),
+        ]
+        #yapf: enable
+
+        all_tested_options_set = set()
+
+        for option in options:
+            all_tested_options_set.add(self._convert_flag(option.long_flag))
+
+            if option.type in ["bool"]:
```

Comment:
using an enum here would make sure that you are covering all the cases via an 'else: assert'

Reply:
I can always just add an assert here as an else anyway. If we switch to using an enum, I will update this part too.